src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1123,6 +1102,11 @@ _filter_helper_astable(f, nti::Tables.NamedTupleIterator)::BitVector = (x -> f(x
 
 Remove rows from data frame `df` for which `fun` returns `false`.
 
+This method is defined so that DataFrames.jl implements the Julia API for
+collections, but it is generally recommended to use the [`subset!`](@ref)
+function instead, as it is both faster and more consistent with other
```

Comment:
Actually I forgot that you could pass `cols` here to ensure type stability, and `subset` isn't faster in that case, right? Maybe we should only mention consistency?

Also, I'd only mention this at the end of the docstring, possibly in a `!!! note`. Better not add too much noise for people who do want to use that function.

Reply:
changed all docstrings for `filter` and `subset` and the manual entry.